lib/providers/password.js

```diff
@@ -16,31 +16,46 @@ var errors = require('../errors');
 function PasswordProvider (client, options) {
     // Jar for storing the session cookie
     this.jar = request.jar();
+    // storage of the CSRF Token as given in the response from the server
+    this.csrfToken = null;
+    this.csrfTokenLocation = 'x-csrf-token';
```

Comment:
Why store this in each instance? I feel like you can hardcode this or make it a constant (var top of file) instead.